app/src/ereferrals/java/org/eyeseetea/malariacare/data/sync/importer/metadata/configuration/MetadataConfigurationDataSourceFactory.java

```diff
@@ -2,16 +2,15 @@
 
 import android.support.annotation.NonNull;
 
-import org.eyeseetea.malariacare.data.database.utils.PreferencesState;
 import org.eyeseetea.malariacare.data.remote.IMetadataConfigurationDataSource;
 import org.eyeseetea.malariacare.network.retrofit.BasicAuthInterceptor;
 
 public class MetadataConfigurationDataSourceFactory {
     @NonNull
     public static IMetadataConfigurationDataSource getMetadataConfigurationDataSource(
-            BasicAuthInterceptor basicAuthInterceptor)
+            BasicAuthInterceptor basicAuthInterceptor, String url)
```

Comment:
We should get that preference from SettingRepository.
I move the code to get the program url from SettingsDataSource to PreferencesEReferal to avoid duplicate code, and i used it in getMetadataConfigurationDataSource method.
I needed add a MetadataConfigurationDataSource  constructor to pass the context to get the preference. (the alternative is PreferenceState.getInstance().getContext()).